src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -476,6 +476,10 @@ private void updateScore(Result result, Set<ProgrammingExerciseTestCase> success
         }
         else {
             double weightSum = allTests.stream().filter(testCase -> !testCase.isInvisible()).mapToDouble(ProgrammingExerciseTestCase::getWeight).sum();
+            if (Math.abs(weightSum - 0) < 1E-8) {
```

Comment:
Why exactly do you need the "- 0" though?

Reply:
@SimonEntholzer the `-0` has no effect but makes it easy to understand that there's a floating point comparison:

If you use the pattern `Math.abs(x - VAL) < EPS` for any `VAL`, you can easily see that you compare a variable to this specific value `VAL` :)

Reply:
this code looks too magic, can we please add a short comment what it is supposed to do?